converter/dicom-cast/src/Microsoft.Health.DicomCast.TableStorage/Registration/DicomCastTableRegistrationExtension.cs

```diff
@@ -61,19 +61,19 @@ private static IServiceCollection AddTableDataStore(this IServiceCollection serv
                 .Singleton()
                 .AsSelf();
 
-            serviceCollection.Add<TableClientProvider>()
+            serviceCollection.Add<TableServiceClientProvider>()
                 .Singleton()
                 .AsSelf()
                 .AsService<IHostedService>()
                 .AsService<IRequireInitializationOnFirstRequest>();
 
-            serviceCollection.Add(sp => sp.GetService<TableClientProvider>().CreateTableClient())
+            serviceCollection.Add(sp => sp.GetService<TableServiceClientProvider>().CreateTableServiceClient())
                 .Singleton()
                 .AsSelf();
 
-            serviceCollection.Add<TableClientInitializer>()
```

Comment:
You may want to use the registration methods that I think are from `Azure.Core`. That way the clients leverage the loggers from our dependency injection container. You can use the extension method `AddAzureClients` on an `IServiceCollection` and add your table client within the configure delegate. I did something similar for blob clients in healthcare shared components [here](https://github.com/microsoft/healthcare-shared-components/blob/713e0357a06ac697eda616cd9d0486089bb1efbd/src/Microsoft.Health.Blob/Registration/BlobClientRegistrationExtensions.cs#L124).

If you don't need to special-case the token credential and can let the built-in factories create it, you can probably invoke something like `AddTableServiceClient` with an `IConfiguration` containing both the authentication and the table info.